packages/internal-testing-tools/android/src/main/kotlin/com/datadog/reactnative/internaltesting/DdInternalTestingImplementation.kt

```diff
@@ -0,0 +1,240 @@
+import com.datadog.android.api.context.DatadogContext
+import com.datadog.android.api.context.NetworkInfo
+import com.datadog.android.api.context.TimeInfo
+import com.datadog.android.api.feature.Feature
+import com.datadog.android.api.feature.FeatureScope
+import com.datadog.android.api.storage.EventBatchWriter
+import com.datadog.android.api.storage.RawBatchEvent
+import com.datadog.android.core.InternalSdkCore
+import com.datadog.android.core.internal.net.FirstPartyHostHeaderTypeResolver
+import com.datadog.android.trace.TracingHeaderType
+import com.datadog.reactnative.DatadogSDKWrapperStorage
+import com.facebook.react.bridge.Promise
+import com.google.gson.Gson
+import okhttp3.HttpUrl
+import kotlin.concurrent.thread
+
+/**
+ * The entry point to use Datadog's internal testing feature.
+ */
+class DdInternalTestingImplementation() {
+    private var wrappedCore: StubSDKCore? = null
+
+    /**
+     * Clears all data for all features.
+     */
+    fun clearData(promise: Promise) {
+        wrappedCore?.clearData()
+        promise.resolve(null)
+    }
+
+    /**
+     * Retrieves the list of events for a given feature.
+     */
+    fun getAllEvents(feature: String, promise: Promise) {
+        val events = wrappedCore?.eventsWritten(feature)
+        val eventsJson = Gson().toJson(events)
+        promise.resolve(eventsJson)
+    }
+
+    /**
+     * Enable native testing module.
+     */
+    fun enable(promise: Promise) {
+        DatadogSDKWrapperStorage.addOnInitializedListener { ddCore ->
+            ddCore?.let { core ->
+                this.wrappedCore = StubSDKCore(core)
+                DatadogSDKWrapperStorage.setSdkCore(this.wrappedCore)
+            }
+        }
+        promise.resolve(null)
+    }
+
+
+    companion object {
+        internal const val NAME = "DdInternalTesting"
+    }
+}
+
+internal class StubSDKCore(
+    private val core: InternalSdkCore
+) : InternalSdkCore by core {
+    internal val featureScopes = mutableMapOf<String, FeatureScopeInterceptor>()
+
+    // region Stub
+
+    /**
+     * Lists all the events written by a given feature.
+     * @param featureName the name of the feature
+     * @return a list of [StubEvent]
+     */
+    fun eventsWritten(featureName: String): List<String> {
+        return featureScopes[featureName]?.eventsWritten()?.toList() ?: emptyList<String>()
+    }
+
+    fun clearData() {
+        featureScopes.forEach { entry ->
+            featureScopes[entry.key]?.clearData()
+        }
+    }
+
+    // endregion
+
+    // region InternalSdkCore
+
+    override val firstPartyHostResolver: FirstPartyHostHeaderTypeResolver =
+        StubFirstPartyHostHeaderTypeResolver()
+
+    override fun getDatadogContext(): DatadogContext? {
+        return core.getDatadogContext()
+    }
+
+    override val networkInfo: NetworkInfo
+        get() = core.networkInfo
+
+    // endregion
+
+    // region FeatureSdkCore
+
+    override val internalLogger: InternalLogger = core.internalLogger
```

Comment:
all the calls which are calling the same method/property of `core` are not needed, they will be automatically handled by using `by` delegate.